docs/development/releasing/index.md

```diff
@@ -2,41 +2,16 @@
 
 ## Helm Charts
 
-### 1. Build and test locally:
+**1.** Upgrade to the latest cluster component versions:
```

Comment:
I thought of keeping it simple, one topic per section. We could add other ones e.g. `## Docker images`, `## Python packages` etc... what do you think?

Reply:
Yeah, we'd need to add all the other section if we want to have the full release instruction picture... The chain is quite complex.

Reply:
I've included the simplest steps for every category. Ideally, we should migrate https://digital-repositories.web.cern.ch/digital-repositories/reana/releasing/ here. After having a look to it, I have the following comments:
- Shall we move [what is a REANA release](https://digital-repositories.web.cern.ch/digital-repositories/reana/releasing/#what-is-a-reana-release) here, with the same name or rename as "Goal"?
- It would be helpful to move the table that shows where to release what (see [here](https://digital-repositories.web.cern.ch/digital-repositories/reana/releasing/#release))
- It would also be helpful to document the release order
    - Arguably, this is already documented now, as first we show Python package releases, then Docker and then Helm. To know which component is which (whether Python package, Docker or Helm) one can look up the previous table
- We would be missing the `reana-dev release-github`, if we want to keep the [release on GitHub](https://digital-repositories.web.cern.ch/digital-repositories/reana/releasing/#github).
    - So far we were releasing every component/package on GitHub, however, I don't think it is necessary as long as `reanahub/reana` is released on GitHub, which it is by using `reana-dev release-helm`
- Should [after release steps](https://digital-repositories.web.cern.ch/digital-repositories/reana/releasing/#after-release) be documented here? (follow get started again, forum and twitter posts....)

Reply:
Hmm, I'm not sure it would be of interest to `docs.reana.io` web site visitors to see so much details... if it pops up in searching for usage guide for example... So if we do port it, then it had better be somehow separated from the user content, a bit "hidden" in another search index, or perhaps even presented in a different coloured style... So that people wont' be confused that when they search for helm (aiming to see how to install REANA on GCP), they would end up on a page explaining how to create Helm charts themselves.  The end user researchers installing REANA on their laptops could get really confused.

(Example: I'm not sure whether in the Kubernetes documentation one can find a corner that speaks about how to release Kubernetes itself? That would confuse quite a few users... Better keep it separated.)

I'm not sure how this Releasing section began here, but perhaps we could keep it separated? Either on DR docs, either on GitHub wiki... somewhere that it is not searched in common.  Or keep here but explore putting it under different section/search index? WDYT?

P.S. A minor comment about GitHub releases: this could indeed stay at the level of pushed tags only, especially for alpha/beta releases.  However, for production releases, we may want to populate the release news... Could be manually (since we don't do many of these per year yet), could be automated later via `release-github` indeed!

Reply:
@diegodelemos What about removing the "Releasing" section from here and moving the content to `reana`'s GitHub Wiki, as we discussed IRL last week?

(Also, needs a rebase against latest master, but let's perhaps merge the other pending PR first.)

Reply:
Moved to GitHub wiki https://github.com/reanahub/reana/wiki/Releasing.